### PR TITLE
callback to transition to ready state 5 seconds after the car faults

### PR DESCRIPTION
**Pull request by Sabramz**
## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: Core/Src/fault.c [hunk 1]
```diff
@@ -28,31 +28,45 @@ int queue_fault(fault_data_t* fault_data)
...
+void can_send_fault_msg(fault_data_t* fault_data) {
+	uint32_t fault_id = (uint32_t) fault_data->id;
+	endian_swap(&fault_id, 4);
+	uint8_t defcon = (uint8_t) fault_data->severity;
+
+	can_msg_t msg;
+	msg.id = CANID_FAULT_MSG;
+	msg.len = 8;
+	uint8_t msg_data[8];
+	memcpy(msg_data, &fault_id, 4);
```
Comment (nwdepatie) [inline Core/Src/fault.c:40]: lmao why do we memcpy this 3 times

Comment (nwdepatie) [reply Core/Src/fault.c:40]: we could just bitwise pack a struct if needed

File: Core/Src/fault.c [hunk 2]
```diff
@@ -28,31 +28,45 @@ int queue_fault(fault_data_t* fault_data)
...
-			msg.id = CANID_FAULT_MSG;
-			msg.len = 8;
-			uint8_t msg_data[8];
-			memcpy(msg_data, &fault_id, 4);
-			memcpy(msg_data + 4, &defcon, 1);
-			memcpy(msg.data, msg_data, msg.len);
-			queue_can_msg(msg);
+			can_send_fault_msg(&fault_data);
 			printf("\r\nFault Handler! Diagnostic Info:\t%s\r\n\r\n", fault_data.diag);
+			osTimerStart(fault_timer, 5000);
```
Comment (nwdepatie) [inline Core/Src/fault.c:69]: This timer should exist within state machine. Logically, queue fault will just evaluate the severity of the fault and elevate the fault in the event that we need to shut off car, but state transition is the thing that actually performs the operations

**Review by nwdepatie — CHANGES_REQUESTED** [review]
Fix misc comments we can chat

File: Core/Src/state_machine.c [hunk 3]
```diff
@@ -97,7 +97,6 @@ int queue_state_transition(state_req_t new_state)
 				write_fan_battbox(pdu, true);
 				write_pump(pdu, false);
 				write_fault(pdu, false);
```
Comment (Sabramz) [inline Core/Src/state_machine.c:99]: Do I need to put the timer here as well?
